commons/proc/nfsd.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2014 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proc
+
+import (
+	"github.com/zenoss/glog"
+
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+)
+
+var procNFSDExportsFile = "fs/nfsd/exports"
+
+func GetProcNFSDExportsFilePath() string {
+	return fmt.Sprintf("%s%s", procDir, procNFSDExportsFile)
+}
+
+var ErrMountPointNotExported = fmt.Errorf("mount point not exported")
+
+// ProcNFSDExports is a parsed representation of /proc/fs/nfsd/exports information.
+type ProcNFSDExports struct {
+	MountPoint    string                       // exported path on the server: /exports/serviced_var
+	ClientOptions map[string]NFSDExportOptions // keys are clients: 'Machine Name Formats' of exports manpage
+}
+
+// Equals checks the equality of two ProcNFSDExports
+func (a *ProcNFSDExports) Equals(b *ProcNFSDExports) bool {
+	if a.MountPoint != b.MountPoint {
+		return false
+	}
+	if !reflect.DeepEqual(a.ClientOptions, b.ClientOptions) {
+		return false
+	}
+	return true
+}
+
+// NFSDExportOptions are options specified in 'General Options' of exports manpage
+type NFSDExportOptions map[string]string
+
+// GetProcNFSDExport gets the export info to the mountpoint entry in /proc/fs/nfsd/exports
+func GetProcNFSDExport(mountpoint string) (*ProcNFSDExports, error) {
+	exports, err := GetProcNFSDExports()
+	if err != nil {
+		return nil, err
+	}
+
+	export, ok := exports[mountpoint]
+	if !ok {
+		return nil, ErrMountPointNotExported
+	}
+
+	return &export, nil
+}
+
+// GetProcNFSDExports gets a map to the /proc/fs/nfsd/exports
+func GetProcNFSDExports() (map[string]ProcNFSDExports, error) {
+	// read in the file
+	data, err := ioutil.ReadFile(GetProcNFSDExportsFilePath())
+	if err != nil {
+		return nil, err
+	}
+
+	exports := make(map[string]ProcNFSDExports)
+	scanner := bufio.NewScanner(strings.NewReader(string(data)))
```

Comment:
I would rather not change that at this critical juncture - this is mimicking all the other functions